ros/src/i2c_com/scripts/i2c.py

```diff
@@ -1,17 +1,48 @@
 #! /usr/bin/python
 import rospy
-from shared_msgs.msg import i2c_msg
+import time
+import smbus
+from BNO055 import BNO055
+from TYS01 import TSYS01
+from ms5837 import MS5837
+from shared_msgs.msg import imu_msg
+from shared_msgs.msg import temp_msg
+from shared_msgs.msg import depth_msg
+
+IMU_PUB_RATE = 50
+TEMP_RATIO_RATE = 4
 
 def message_received(msg):
   # This runs on a seperate thread from the pub
   pass
 
 if __name__ == "__main__":
   rospy.init_node('i2c_node')
-  pub = rospy.Publisher('i2c_rx', i2c_msg,
-      queue_size= 100)
-  sub = rospy.Subscriber('i2c_tx', i2c_msg,
-      message_received)
-
-  # TODO: I2C related activities
-  rospy.spin()
+  imu_pub = rospy.Publisher('imu_data', imu_msg, qeue_size = 100)
+  imu_pub.rate(IMU_PUB_RATE)
+  temp_pub = rospy.Publisher('temp_data', temp_msg, qeue_size = 100)
+  depth_pub = rospy.Publisher('depth_data', depth_msg, qeue_size = 100)
+  imu = BNO055()
```

Comment:
Add catches to these sensor constructor calls that will, if they throw a Runtime Exception or something like that, you replace the sensor with a mock (and a print out explaining that the sensor failed to be initialized).

Furthermore, you will have to make mock classes, not that difficult. Just fill the interfaces and return a sine wave or something like that.

Reply:
Also not a big fan of just calling the sensors imu, temp, and depth. I'd consider appending at least _sens or something at the end.